cmd/clusterctl/pkg/client/repository/repository_local_unix.go

```diff
@@ -0,0 +1,215 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package repository
+
+import (
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/version"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/pkg/client/config"
+)
+
+// localRepository provides support for providers located on the local filesystem.
+// As part of the provider object, the URL is expected to contain the absolute
+// path to the components yaml on the local filesystem.
+// In order to support different versions, the directories containing provider
+// specific data must adhere to the following layout:
+// {basepath}/{owner}/{provider-name}/releases/{version}/{path/components.yaml}
```

Comment:
Considering that we are already under a specific {basepath}, I think we can avoid the {path} bit.
Additionally, if implemented, this change improves the consistency of the user experience, because it makes the directory structure of the file system provider to match the directory structure of the local override folder

Reply:
Done.